### PR TITLE
subsys: fs: nvs: Migrate to new logging subsys

### PR DESCRIPTION
**Pull request by Olivier-ProGlove**
Migrate from `SYS_LOG` to `LOG` logging mechanism.

Signed-off-by: Olivier Martin <olivier.martin@proglove.de>

### CODE REVIEW

File: subsys/fs/nvs/nvs_priv.h [hunk 1]
```diff
@@ -11,9 +11,9 @@
 extern "C" {
 #endif
 
-#define SYS_LOG_DOMAIN "fs/nvs"
+#define LOG_MODULE_NAME fs_nvs
```
Comment (nordic-krch) [inline subsys/fs/nvs/nvs_priv.h]: i would move this block to nvs.c next to module registration.
Add:
`#define LOG_LEVEL CONFIG_NVS_LOG_LEVEL`
To get logging configuration from kconfig.

**Review by nordic-krch — CHANGES_REQUESTED** [review]

**Comment by nvlsianpu** [comment]
@Olivier-ProGlove Thanks for this effort!
@nordic-krch, @carlescufi   Is it possible that we first merge #8141 and then replay this PR o that?

**Comment by nvlsianpu** [comment]
@Olivier-ProGlove Can you rework this PR as PR #8141 was merged. I belief that now the mainline code is almost there.

**Comment by Olivier-ProGlove** [comment]
@nvlsianpu Pull-request rebased
@nordic-krch Comment taken in account

**Review by nvlsianpu — APPROVED** [review]

File: subsys/fs/nvs/nvs_priv.h [hunk 2]
```diff
@@ -11,9 +11,7 @@
 extern "C" {
 #endif
 
-#define SYS_LOG_DOMAIN "fs/nvs"
-#define SYS_LOG_LEVEL CONFIG_NVS_LOG_LEVEL
-#include <logging/sys_log.h>
+#include <logging/log.h>
```
Comment (nordic-krch) [inline subsys/fs/nvs/nvs_priv.h]: this include can also be moved to nvs.c.

**Comment by nashif** [comment]
addressed in #10029

**Pull request closed without merging**